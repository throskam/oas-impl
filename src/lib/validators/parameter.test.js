const parameterValidator = require('./parameter')

describe('Required', () => {
  const parameter = {
    name: 'param',
    required: true
  }

  const validator = parameterValidator(parameter)

  it('should return no errors when the value is defined', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a required error when the value is missing', () => {
    expect(validator()).toMatchSnapshot()
  })
})

describe('Optional', () => {
  const parameter = {}
  const validator = parameterValidator(parameter)

  it('should return no errors when the value is defined', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return no errors when the value is missing', () => {
    expect(validator()).toMatchSnapshot()
  })
})

describe('Schema', () => {
  const parameter = {
    name: 'param',
    required: true,
    schema: {
      type: 'integer'
    }
  }

  const validator = parameterValidator(parameter)

  it('should return no errors when the value is valid', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is invalid', () => {
    const payload = { value: 'string' }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is null', () => {
    const payload = { value: null }
    expect(validator(payload)).toMatchSnapshot()
  })
})

describe('Content', () => {
  const parameter = {
    name: 'param',
    required: true,
    content: {
      'application/json': {
        schema: {
          type: 'integer'
        }
      }
    }
  }

  const validator = parameterValidator(parameter)

  it('should return no errors when the value is valid', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is invalid', () => {
    const payload = { value: 'string' }
    expect(validator(payload)).toMatchSnapshot()
  })
})

describe('Deprecated', () => {
  const parameter = {
    name: 'param',
    deprecated: true,
    allowEmptyValue: true,
    schema: {
      type: 'integer'
    }
  }

  const validator = parameterValidator(parameter)

  it('should return no errors when the value is missing', () => {
    expect(validator()).toMatchSnapshot()
  })

  it('should return a deprecated error when the value is defined', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a deprecated error when the value is null', () => {
    const payload = { value: null }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a deprecated error and validation errors when the value is defined and invalid', () => {
    const payload = { value: 'string' }
    expect(validator(payload)).toMatchSnapshot()
  })
})

describe('Allow empty value', () => {
  const parameter = {
    name: 'param',
    required: true,
    allowEmptyValue: true,
    schema: {
      type: 'integer'
    }
  }

  const validator = parameterValidator(parameter)

  it('should return no errors when the value is null', () => {
    const payload = { value: null }
    expect(validator(payload)).toMatchSnapshot()
  })
})
