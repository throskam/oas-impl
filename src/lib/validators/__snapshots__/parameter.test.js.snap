// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allow empty value should return no errors when the value is null 1`] = `Array []`;

exports[`Content should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "parameter[param].content[application/json].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Content should return no errors when the value is valid 1`] = `Array []`;

exports[`Deprecated should return a deprecated error and validation errors when the value is defined and invalid 1`] = `
Array [
  Object {
    "path": "parameter[param]",
    "rule": "parameter-deprecated",
    "value": "string",
  },
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "parameter[param].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Deprecated should return a deprecated error when the value is defined 1`] = `
Array [
  Object {
    "path": "parameter[param]",
    "rule": "parameter-deprecated",
    "value": 1,
  },
]
`;

exports[`Deprecated should return a deprecated error when the value is null 1`] = `
Array [
  Object {
    "path": "parameter[param]",
    "rule": "parameter-deprecated",
    "value": null,
  },
]
`;

exports[`Deprecated should return no errors when the value is missing 1`] = `Array []`;

exports[`Optional should return no errors when the value is defined 1`] = `Array []`;

exports[`Optional should return no errors when the value is missing 1`] = `Array []`;

exports[`Required should return a required error when the value is missing 1`] = `
Array [
  Object {
    "path": "parameter[param]",
    "rule": "parameter-required",
  },
]
`;

exports[`Required should return no errors when the value is defined 1`] = `Array []`;

exports[`Schema should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "parameter[param].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Schema should return a validation error when the value is null 1`] = `
Array [
  Object {
    "path": "parameter[param]",
    "rule": "parameter-deny-empty-value",
  },
]
`;

exports[`Schema should return no errors when the value is valid 1`] = `Array []`;
